coalib/output/TomlConfWriter.py

```diff
@@ -0,0 +1,177 @@
+        if sections.get('cli', None):
+            self.remove_unsavable_settings(sections)
+
+        for item in sections:
+            section = sections[item]
+            table_name = self.get_table_name(section)
+            if table_name in self.document:
+                continue
+            table_contents = table()
+            for _, setting in section.contents.items():
+
+                setting_key = self.get_setting_key(setting)
+
+                if isinstance(setting, TomlSetting):
+                    value = setting.original_value
+                else:
+                    value = self.get_original_value(setting.value)
+
+                if isinstance(value, Array):
+                    table_contents.add(setting_key, array(value.as_string()))
+                    if value.trivia.comment:
+                        table_contents[setting_key].comment(value.trivia
+                                                            .comment)
+                elif isinstance(value, String):
+                    table_contents.add(setting_key, string(value))
+                    if value.trivia.comment:
+                        table_contents[setting_key].comment(value.trivia
+                                                            .comment)
+                elif isinstance(value, Bool):
+                    table_contents.add(setting_key, value)
+                elif isinstance(value, Integer):
+                    table_contents.add(setting_key, integer(value.as_string()))
+                    if value.trivia.comment:
+                        table_contents[setting_key].comment(value.trivia
+                                                            .comment)
+                elif isinstance(value, Comment):
+                    table_contents.add(Comment(
+                        Trivia(comment_ws='  ', comment=str(value))
+                    ))
+                else:
+                    table_contents.add(setting_key, value)
+
+            self.document.add(table_name, table_contents)
+        self.write_to_file()
+
+    def write_to_file(self):
+        """
+        Write the TOML document in the specified file
+        """
+        with open(self.file, 'w') as f:
+            f.write(dumps(self.document))
+
+    def remove_unsavable_settings(self, sections):
+        """
+        Removes the un-savable keys from the sections
+        before they are written into a file
+
+        :param sections: The sections that have to be
+                         written into the file
+        """
+
+        cli_section_contents = sections['cli'].contents
+        for unsavable_key in self.unsavable_keys:
+            if unsavable_key in cli_section_contents:
+                del cli_section_contents[unsavable_key]
+
+    @staticmethod
+    def get_original_value(value):
+        """
+        Converts a string into its original value
+
+        :param value: The value as a string
+        :return: The original value deduced from the string
```

Comment:
`coafile configuration` sees every settings as a single stiring. But toml configuration has types.  So this function tries to infer the datatype from the string for toml.

`Are you sure this will return the expected type everytime?`
It handles strings, booleans and list of strings and integers. 
These are the settings types  I have seen used by coala . If more types are used like floats, then then this function can make a provision for that as well. 
So I think for now it returns the expected types.